bin/check-all.sh

```diff
@@ -60,9 +60,9 @@ fi
 if [ -e "${base_dir}/gradlew" ]; then
   gradle_file="${base_dir}/gradlew"
 else
-  gradle_base_file=$(which gradle)
+  gradle_base_file=$(which gradle) && chk_gradle=$? || chk_gradle=$?
```

Comment:
`chk_gradle=$?`

Is this is just a way of checking `chk_gradle` when `set -e` is set? Otherwise it would cause the shell to exit if any subcommand or pipeline returns a non-zero status?

Wondering what's the need for this additional check? It seems that `-f`  should already do the trick? (if `gradle_base_file` does not exist)